src/Utilities/model.jl

```diff
@@ -14,14 +14,21 @@ function MOI.get(model::AbstractModel, ::MOI.NumberOfVariables)::Int64
     end
 end
 
+# Use `-Inf` and `Inf` for `AbstractFloat` subtypes.
+_no_lower_bound(::Type{T}) where {T} = zero(T)
+_no_lower_bound(::Type{T}) where {T<:AbstractFloat} = typemin(T)
+_no_upper_bound(::Type{T}) where {T} = zero(T)
+_no_upper_bound(::Type{T}) where {T<:AbstractFloat} = typemax(T)
+
 function MOI.add_variable(model::AbstractModel{T}) where {T}
     vi = VI(model.num_variables_created += 1)
     push!(model.single_variable_mask, 0x0)
-    push!(model.lower_bound, zero(T))
-    push!(model.upper_bound, zero(T))
+    push!(model.lower_bound, _no_lower_bound(T))
+    push!(model.upper_bound, _no_upper_bound(T))
     if model.variable_indices !== nothing
         push!(model.variable_indices, vi)
     end
+    _add_variable(model.constraints)
```

Comment:
It seems weird to force other types to overload `_add_variable`. Why not just overload `MOI.add_variable(model.constraints)`?

Reply:
Then `VectorOfConstraints` should return a `VariableIndex` but currently it does not keep track of this information.